### PR TITLE
feat(blueprints-cli): exporting projects as bundles [experimental]

### PR DESCRIPTION
**Pull request by alexforsyth**
### Issue

Bundles are transportable formats of the contents of a project. 

Blueprints generate bundles which are then interpreted to seed (or modify) the contents of a project. We can also generate bundles from projects themselves. This PR allows anyone to create a bundle from their codecatalyst project.

This is particularly useful for testing resynthesis locally and importing/exporting project between spaces.

### Notes
This CLI tool relies on APIs that may change in backwards incompatible ways in the future. For that reason thes functions are marked EXPERIMENTAL. Do not rely on this code in your own services

### Usage

Example Usage:
```
yarn blueprint bundle-project --project my-awesome-project --space my-awesome-space
```

### CODE REVIEW

File: packages/utils/blueprint-cli/src/bundle/get-bundle-aws-account-to-environment.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      input: {
+        spaceName: options.spaceName,
+        projectName: options.projectName,
+        environmentName: options.environmentName,
+        nextToken: options.nextToken,
+      },
+    },
+  });
+
+  const response = await executeRequest<ApolloQueryResult<any>>(logger, listAwsAccountAssociation, {
```
Comment (skeppeler) [inline packages/utils/blueprint-cli/src/bundle/get-bundle-aws-account-to-environment.ts:65]: a nice todo for the future, but something to think about is a version of `executeRequest` that supports pagination tokens.

File: packages/utils/blueprint-cli/src/bundle/package-bundle.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+export function packageBundle(
+  logger: pino.BaseLogger,
+  folderPathAbs: string,
+  outputPathAbs: string,
+  options: {
+    encrypt?: boolean;
+  },
+) {
+  try {
+    let zipCommand = [`zip ${path.basename(outputPathAbs)}`, `-r ${path.basename(folderPathAbs)}`, '-x **/.git/**'].join(' ');
```
Comment (skeppeler) [inline packages/utils/blueprint-cli/src/bundle/package-bundle.ts:17]: `tar` might provide better interoperability.

File: packages/utils/blueprint-cli/src/bundle/write-elements.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+function writeYaml(logger: pino.BaseLogger, filePathAbs: string, content: any) {
+  logger.debug(`Writing: ${filePathAbs}`);
+  const yamlcontent = yaml.stringify(content);
+  console.log(yamlcontent);
+  fs.mkdirSync(path.dirname(filePathAbs), { recursive: true });
+  fs.writeFileSync(filePathAbs, yaml.stringify(content));
+}
+
+function stripName(name: string): string {
+  return name;
```
Comment (skeppeler) [inline packages/utils/blueprint-cli/src/bundle/write-elements.ts:69]: Should this be filtering out special characters or something?

**Review by skeppeler — APPROVED** [review]

File: packages/utils/blueprint-cli/src/bundle/package-bundle.ts (hunk 2 above)
Comment (alexforsyth) [reply packages/utils/blueprint-cli/src/bundle/package-bundle.ts:17]: Tar unfortunately does not support an native encryption standard. Which practically makes it kinda annoying to use. 

The idea here is that your password can be used if you manually double click to unzip your zip file.

File: packages/utils/blueprint-cli/src/bundle/get-bundle-aws-account-to-environment.ts (hunk 1 above)
Comment (alexforsyth) [reply packages/utils/blueprint-cli/src/bundle/get-bundle-aws-account-to-environment.ts:65]: ExecutePaginatedRequest?

Comment (alexforsyth) [reply packages/utils/blueprint-cli/src/bundle/get-bundle-aws-account-to-environment.ts:65]: Yeah thats a good point, i'll add one in the future if needed.

File: packages/utils/blueprint-cli/src/bundle/write-elements.ts (hunk 3 above)
Comment (alexforsyth) [reply packages/utils/blueprint-cli/src/bundle/write-elements.ts]: updated

**Pull request merged**